ndcube/ndcube.py

```diff
@@ -652,6 +654,85 @@ def explode_along_axis(self, axis):
+        Reprojects this NDCube to the coordinates described by another WCS object.
+
+        Parameters
+        ----------
+        target_wcs : `astropy.wcs.wcsapi.BaseHighLevelWCS`, `astropy.wcs.wcsapi.BaseLowLevelWCS`,
+            or `astropy.io.fits.Header`
+            The WCS object on which the ``NDCube`` is to be reprojected.
+
+        shape_out: `tuple`, optional
+            The shape of the output data. If not specified, `~astropy.wcs.wcsapi.BaseLowLevelWCS.pixel_shape` attribute (if available)
+            from the low level API of the ``target_wcs`` is used.
+
+        order: `int` or `str`
+            The order of the interpolation. This can be any of: 'nearest-neighbour', 'bilinear',
+            'biquadratic', or 'bicubic'.
+
+        output_array: `numpy.ndarray`, optional
+            An array in which to store the reprojected data. This can be any numpy array
+            including a memory map, which may be helpful when dealing with extremely large files.
+
+        return_footprint: `bool`
+            Whether to return the footprint in addition to the output array.
+
+        Returns
+        -------
+        resampled_cube : `ndcube.NDCube`
+            A new resultant NDCube object, the supplied ``target_wcs`` will be the ``.wcs`` attribute of the output ``NDCube``.
+
+        footprint: `numpy.ndarray`
+            Footprint of the input array in the output array. Values of 0 indicate no coverage or
+            valid values in the input image, while values of 1 indicate valid values.
+
+        Notes
+        -----
+        This method doesn't support handling of the ``mask``, ``extra_coords``, and ``uncertainty`` attributes yet.
+        However, ``meta`` and ``global_coords`` are copied to the output ``NDCube``.
+        """
+        try:
+            from reproject import reproject_interp
+        except ModuleNotFoundError:
+            raise ImportError("This method requires the optional package `reproject`.")
+
+        if isinstance(target_wcs, Mapping):
+            try:
+                target_wcs = WCS(header=target_wcs)
+            except Exception:
+                raise Exception('Invalid header supplied; unable to create a WCS object.')
+
+        if not utils.wcs.compare_wcs_physical_types(self.wcs, target_wcs):
+            raise Exception('Given target_wcs is not compatible with this NDCube, the physical types do not match.')
+
+        low_level_target_wcs = utils.wcs.get_low_level_wcs(target_wcs, 'target_wcs')
+
+        # If shape_out is not specified explicity, try to extract it from the low level WCS
+        if not shape_out:
+            if hasattr(low_level_target_wcs, 'pixel_shape') and low_level_target_wcs.pixel_shape is not None:
+                shape_out = low_level_target_wcs.pixel_shape
+            else:
+                raise Exception("shape_out must be specified if target_wcs's low level API "
+                                "does not have the pixel_shape attribute.")
```

Comment:
I believe these should all be `array_shape` instead of `pixel_shape`

Reply:
Yes, looking back at it, I think it should be `array_shape`. Will change it.